src/diskcryptor_common.h

```diff
@@ -27,6 +27,17 @@ struct custom_salt {
 #define PKCS_DERIVE_MAX         (MAX_KEY_SIZE*2)
 #define CF_CIPHERS_NUM          7  // 7 types of ciphers possible
 
+#ifdef __GNUC__
+#define PACKED __attribute__ ((__packed__))
+#else
+#define PACKED
+#endif
+
+#ifndef __GNUC__
+#define PACKED
+#pragma pack(push,1)
+#endif
```

Comment:
Perhaps move the lower one of these to the else clause of the upper one?

Reply:
I am not about this but perhaps the `#pragma pack(pop)` part needs to come after the structure definition (and thus needs to belong to a new `ifdef` block)?